src/app/buttons/button-loading.ts

```diff
@@ -4,35 +4,45 @@
  * The full license information can be found in LICENSE in the root directory of this project.
  */
 import {Component} from "@angular/core";
+import {ClrLoadingState} from "../../clr-angular";
+
 
 @Component({
     selector: "clr-buttons-demo-button-loading",
     templateUrl: "./button-loading.html",
     styleUrls: ["./buttons.demo.scss"]
 })
 export class ButtonLoadingDemo {
-    private validateLoading: boolean = false;
-    private submitLoading: boolean = false;
-    private validateSmLoading: boolean = false;
-    private submitSmLoading: boolean = false;
+    private validateState: ClrLoadingState = ClrLoadingState.DEFAULT;
+    private submitState: ClrLoadingState = ClrLoadingState.DEFAULT;
+    private validateSmState: boolean = false;
+    private submitSmState: ClrLoadingState = ClrLoadingState.DEFAULT;
 
     validateDemo() {
-        this.validateLoading = true;
-        setTimeout(() => this.validateLoading = false, 1500);
+        this.validateState = ClrLoadingState.LOADING;
+        setTimeout(() => {
+            this.validateState = ClrLoadingState.SUCCESS;
+        }, 1500);
     }
 
     submitDemo() {
-        this.submitLoading = true;
-        setTimeout(() => this.submitLoading = false, 1500);
+        this.submitState = ClrLoadingState.LOADING;
+        setTimeout(() => {
+            this.submitState = ClrLoadingState.DEFAULT;
```

Comment:
What is the difference between changing from `LOADING` to `SUCCESS` like you did in the `validateDemo()` method vs changing from `LOADING` to `DEFAULT`?

Reply:
Nvm. I should have read `loading-button.ts` first :D